tfhe/src/typed_api/shortints/client_key.rs

```diff
@@ -0,0 +1,38 @@
+use std::marker::PhantomData;
+
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "internal-keycache")]
+use crate::shortint::keycache::KEY_CACHE;
+use crate::shortint::ClientKey;
+
+use super::parameters::ShortIntegerParameter;
+
+/// The key associated to a short integer type
+///
+/// Can encrypt and decrypt it.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct GenericShortIntClientKey<P: ShortIntegerParameter> {
+    pub(super) key: ClientKey,
+    _marker: PhantomData<P>,
+}
+
+impl<P> From<P> for GenericShortIntClientKey<P>
+where
+    P: ShortIntegerParameter,
+{
+    fn from(parameters: P) -> Self {
+        #[cfg(feature = "internal-keycache")]
+        let key = KEY_CACHE
+            .get_from_param(parameters.into())
+            .client_key()
+            .clone();
+        #[cfg(not(feature = "internal-keycache"))]
+        let key = ClientKey::new(parameters.into());
```

Comment:
can be a bit dangerous for end users ?

Reply:
What would be dangerous ?

Reply:
the keycache part

Reply:
yes but in what is could it be dangerous ?

When I wrote this in concrete, the `internal-keycache`  was not meant for end users, only for internal development.
in tfhe-rs this feature is considered public ?

Reply:
🤔 I don't how much the faster keygen improves things, but it was necessary to avoid doctests taking hours

Reply:
whole keyset used to take 15 minutes on our big test instances and key gen is about 2 to 4 minutes IIRC, now the biggest keys take 20 seconds I think